karyon3-api/src/main/java/com/netflix/karyon/api/PropertySource.java

```diff
@@ -49,5 +45,5 @@
      * @param defaultValue
      * @return
      */
-    public <T> T get(String key, Class<T> type, T defaultValue);
+    <T> T get(String key, Class<T> type, T defaultValue);
```

Comment:
I know the type system in Java is a little funky sometimes, but could this just be.

<T> T get(String key, T defaultValue); and allow you to get rid of the String version above?